### DISCUSSION TITLE
Is it possible to use copilot with repositories not hosted in Github.com ?

### DISCUSSION THREAD

**Discussion by nelson777** [community/community#53725] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I'm very interested in using copilot but my company has it's own customized Gitlab installation and I'm not allowed to use Github.com to host corporative projects. Is it possible to use copilot and having the project repository in my company's Gitlab site ?

**Comment by davecheney** [comment, marked as answer]
Yes, you can use Copilot with any source code. It doesn't need to be stored in GitHub, it doesn't need to be checked in to source control at all, Copilot will even work on files that haven't been saved.

**Comment by nelson777** [comment]
Great thanks

**Comment by Ranjucap** [comment]
Can GitHub Copilot be used without using GitHub for source control?

**Reply by davecheney** [reply]
https://github.com/orgs/community/discussions/53725#discussioncomment-5725051
